Scripts/RemediationScripts/Remediate-EnableSoftDeleteForBackupVault.ps1

```diff
@@ -0,0 +1,543 @@
+				Write-Host $([Constants]::DoubleDashLine)	
+				return
+			}
+		}
+		else {
+			Write-Host "'Force' flag is provided. Proceeding to enable always-on soft delete on Backup Vault(s) without any prompts..." -ForegroundColor $([Constants]::MessageType.Warning)
+		}
+
+		# List for storing remediated Backup Vaults(s)
+		$backupVaultsRemidiated = @()
+
+		# List for storing skipped Backup Vaults(s)
+		$backupVaultsSkipped = @()
+
+		# Enable always-on soft delete on each Backup Vault
+		# Loop through the list of Backup Vault(s) which needs to be remediated.
+		$backupVaultsWithoutSoftDelete | ForEach-Object {
+			$backupVault = $_
+			try {
+				if (-not $_.SoftDeleteState) {
+                    Update-AzDataProtectionBackupVault -ResourceGroupName $_.ResourceGroupName -VaultName $_.ResourceName -SoftDeleteState "On" -SoftDeleteRetentionDurationInDay $_.SoftDeleteRetentionDurationInDay -ErrorAction Stop                
+                }
+				$backupVaultResource = Update-AzDataProtectionBackupVault -ResourceGroupName $_.ResourceGroupName -VaultName $_.ResourceName -SoftDeleteState "AlwaysOn" -SoftDeleteRetentionDurationInDay $_.SoftDeleteRetentionDurationInDay -ErrorAction Stop   
+
+				if ($backupVaultResource.SoftDeleteState.ToString() -ieq "ALWAYSON") {
+					$backupVaultsRemidiated += $backupVault
+					$logResource = @{}	
+					$logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+					$logResource.Add("ResourceName", ($_.ResourceName))	
+					$logRemediatedResources += $logResource	
+				}
+				else {
+					$backupVaultsSkipped += $backupVault
+					$logResource = @{}	
+					$logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+					$logResource.Add("ResourceName", ($_.ResourceName))
+					$logResource.Add("Reason", "Error configuring Always On for Backup Vault: [$($backupVault)]")            
+					$logSkippedResources += $logResource	
+
+				}
+                
+			}
+			catch {
+				$backupVaultsSkipped += $backupVault
+				$logResource = @{}	
+				$logResource.Add("ResourceGroupName", ($_.ResourceGroupName))	
+				$logResource.Add("ResourceName", ($_.ResourceName))	
+				$logResource.Add("Reason", "Error configuring Always On for Backup Vault")    	
+				$logSkippedResources += $logResource                
+				Write-Host $([Constants]::SingleDashLine)
+			}
+		}
+
+		Write-Host $([Constants]::DoubleDashLine)
+        
+        
+		if ($AutoRemediation) {
+			if ($($backupVaultsRemidiated | Measure-Object).Count -gt 0) {
+				# Write this to a file.
+				# BackupVaultDetailsBackUp
+				$backupVaultsRemediatedFile = "$($backupFolderPath)\RemediatedBackupVaults.csv"
+				$backupVaultsRemidiated | Export-CSV -Path $backupVaultsRemediatedFile -NoTypeInformation
+
+				Write-Host "The information related to Backup Vault(s) where Always On changed has been saved to [$($backupVaultsRemediatedFile)]. Use this file for any roll back that may be required." -ForegroundColor $([Constants]::MessageType.Update)
```

Comment:
Rollback is not supported right?

Reply:
Removed instances of rollback in comments and code